src/util/newconfig/ConfigDefinition.hpp

```diff
@@ -0,0 +1,230 @@
+     */
+    ValueView
+    getValueInArray(std::string_view fullKey, std::size_t index) const;
+
+    /**
+     * @brief Returns the specified Array object from ClioConfigDefinition
+     *
+     * @param prefix The prefix to search config keys from
+     * @return ArrayView with all key-value pairs where key starts with "prefix"
+     */
+    ArrayView
+    getArray(std::string_view prefix) const;
+
+    /**
+     * @brief Checks if a key is present in the configuration map.
+     *
+     * @param key The key to search for in the configuration map.
+     * @return True if the key is present, false otherwise.
+     */
+    bool
+    contains(std::string_view key) const;
+
+    /**
+     * @brief Checks if any key in config starts with "key"
+     *
+     * @param key The key to search for in the configuration map.
+     * @return True if the any key in config starts with "key", false otherwise.
+     */
+    bool
+    startsWith(std::string_view key) const;
+
+    /**
+     * @brief Returns the Array object associated with the specified key.
+     *
+     * @param key The key whose associated Array object is to be returned.
+     * @return The Array object associated with the specified key.
+     */
+    Array const&
+    atArray(std::string_view key) const;
+
+    /**
+     * @brief Returns the size of an Array
+     *
+     * @param prefix The prefix whose associated Array object is to be returned.
+     * @return The size of the array associated with the specified prefix.
+     */
+    std::size_t
+    arraySize(std::string_view prefix) const;
+
+    /**
+     * @brief Returns an iterator to the beginning of the configuration map.
+     *
+     * @return A constant iterator to the beginning of the map.
+     */
+    auto
+    begin() const
+    {
+        return map_.begin();
+    }
+
+    /**
+     * @brief Returns an iterator to the end of the configuration map.
+     *
+     * @return A constant iterator to the end of the map.
+     */
+    auto
+    end() const
+    {
+        return map_.end();
+    }
+
+private:
+    /**
+     * @brief returns the iterator of key-value pair with key fullKey
+     *
+     * @param fullKey Key to search for
+     * @return iterator of map
+     */
+    auto
+    getArrayIterator(std::string_view key) const
+    {
+        auto fullKey = checkForBracketsInArray(key);
+        auto it = std::find_if(map_.begin(), map_.end(), [&fullKey](auto pair) { return pair.first == fullKey; });
+
+        ASSERT(it != map_.end(), "key {} does not exist in config", fullKey);
+        ASSERT(std::holds_alternative<Array>(it->second), "Value of {} is not an array", fullKey);
+
+        return it;
+    }
+
+    /**
+     * @brief Used for all Array API's; check to make sure "[]" exists, if not, append to end
+     *
+     * @param key key to check for
+     * @return the key with "[]" appended to the end
+     */
+    static std::string
+    checkForBracketsInArray(std::string_view key)
```

Comment:
This method is modifying `key`, so I think better would be `addBracketsForArrayKey()`.